src/super_gradients/training/models/model_factory.py

```diff
@@ -132,11 +132,15 @@ def instantiate_model(
         if pretrained_weights is None and num_classes is None:
             raise ValueError("num_classes or pretrained_weights must be passed to determine net's structure.")
 
-        if pretrained_weights and pretrained_weights in PRETRAINED_NUM_CLASSES.keys():
-            num_classes_new_head = core_utils.get_param(arch_params, "num_classes", PRETRAINED_NUM_CLASSES[pretrained_weights])
-            arch_params.num_classes = PRETRAINED_NUM_CLASSES[pretrained_weights]
-        elif pretrained_weights and pretrained_weights is None:
-            raise ValueError(f"Unknown pretrained_weights - couldn't find pretrained weights in {PRETRAINED_NUM_CLASSES.keys()} or platform.")
+        if pretrained_weights:
```

Comment:
Let's update the docstring of this function to explicitly describe the fixed behavior to document this fix